tstate/tstate_test.go

```diff
@@ -831,3 +831,80 @@ func TestUpdatingKeyPermission(t *testing.T) {
 		})
 	}
 }
+
+func TestInsertAllocate(t *testing.T) {
+	tests := []struct {
+		name        string
+		key         string
+		permission1 state.Permissions
+		permission2 state.Permissions
+		shouldFail  bool
+	}{
+		{
+			name:        "key has R then W",
+			key:         "test",
+			permission1: state.Read,
+			permission2: state.Write,
```

Comment:
I don't think this test needs permisison1 and permission2, a single permission should be good enough.